pkg/util/trace_info.go

```diff
@@ -37,6 +37,12 @@ type TraceInfo struct {
 	tempoOrgID          string
 }
 
+// JaegerClient is an interface used to mock the underlying client in tests.
+type JaegerClient interface {
```

Comment:
I slightly wonder if we should make this a generic grpc client interface.  For example, I think it would be good for the project to eventually have an OTEL vulture, which requires a bunch of shuffling around.  The interfaces you've started here are a step in that direction, and so thinking about what that might mean for this interface here would be good too.

Reply:
It's a good point to use Vulture to test other ways to ingest data into Tempo but it may be a premature optimization. The idea of this PR is to increase confidence in the code by adding some test coverage. We can explore further changes later. What do you think @zalegrala?

Reply:
That sounds good to me.  Just something to think about for the future.